### ISSUE TITLE
[tool_crash] ShaderCompilerException

### ISSUE THREAD

**Issue by JumperBot** [flutter/flutter#115433]
## Command
```
flutter run --web-renderer canvaskit -d web-server
```

## Steps to Reproduce
1. Upgrade `Flutter` from f9972818f4e8bf464b378f0942a153fa391a0b7a to 484e09ef798bb6feaad5942b81bcc92baa061e52 using:
```
git fetch
git pull
flutter upgrade
```
2. Run this command on a project that previously worked on f9972818f4e8bf464b378f0942a153fa391a0b7a:
```
flutter run --web-renderer canvaskit -d web-server
```
3. Get frustrated from trying to copy the link after the logs:

## Logs
ShaderCompilerException
```
#0      ShaderCompiler.compileShader (package:flutter_tools/src/build_system/targets/shader_compiler.dart:201:9)
<asynchronous suspension>
#1      writeBundle.<anonymous closure> (package:flutter_tools/src/bundle_builder.dart:185:25)
<asynchronous suspension>
#2      Future.wait.<anonymous closure> (dart:async/future.dart:522:21)
<asynchronous suspension>
#3      writeBundle (package:flutter_tools/src/bundle_builder.dart:163:3)
<asynchronous suspension>
#4      WebDevFS.update (package:flutter_tools/src/isolated/devfs_web.dart:867:9)
<asynchronous suspension>
#5      ResidentWebRunner._updateDevFS (package:flutter_tools/src/isolated/resident_web_runner.dart:490:35)
<asynchronous suspension>
#6      ResidentWebRunner.run.<anonymous closure> (package:flutter_tools/src/isolated/resident_web_runner.dart:284:41)
<asynchronous suspension>
#7      asyncGuard.<anonymous closure> (package:flutter_tools/src/base/async_guard.dart:111:24)
<asynchronous suspension>
```
```
[32m[✓][39m Flutter (Channel master, 3.6.0-3.0.pre.55, on Debian GNU/Linux 11 (bullseye)
    5.10.136-19394-g7a24dee39fa0, locale en_US.UTF-8)
    [32m•[39m Flutter version 3.6.0-3.0.pre.55 on channel master at Removed_Directory_Name_Sorry/flutter
    [32m•[39m Upstream repository https://github.com/flutter/flutter.git
    [32m•[39m Framework revision 484e09ef79 (70 minutes ago), 2022-11-16 05:18:10 -0500
    [32m•[39m Engine revision 0241f18cbe
    [32m•[39m Dart version 2.19.0 (build 2.19.0-406.0.dev)
    [32m•[39m DevTools version 2.19.0

[32m[✓][39m Chrome - develop for the web
    [32m•[39m CHROME_EXECUTABLE = /usr/bin/garcon-url-handler

[33m[!][39m Flutter IDE Support (No supported IDEs installed)
    [32m•[39m IntelliJ - https://www.jetbrains.com/idea/
    [32m•[39m Android Studio - https://developer.android.com/studio/
    [32m•[39m VS Code - https://code.visualstudio.com/

[32m[✓][39m Connected device (2 available)
    [32m•[39m Web Server (web) • web-server • web-javascript • Flutter Tools
    [32m•[39m Chrome (web)     • chrome     • web-javascript • unknown

[32m[✓][39m HTTP Host Availability
    [32m•[39m All required HTTP hosts are available

[33m![39m Doctor found issues in 1 category.

```

## Notes

This line doesn't matter since I am making it use localhost instead:
```
CHROME_EXECUTABLE = /usr/bin/garcon-url-handler
```

Also, I already tried re-cloning `Flutter` already, same issue.

Other than that here's my `Flutter` config:

```
Settings:
  android-sdk: /home/csg/android_sdk/
  enable-linux-desktop: false
  enable-macos-desktop: false
  enable-windows-desktop: false
  enable-android: false
  enable-ios: false
  enable-fuchsia: false
  enable-custom-devices: false

Analytics reporting is currently disabled.
```

Lastly, here's the command output once `verbose` is toggled:

```
flutter run --web-renderer canvaskit -d web-server -v
[ +125 ms] executing: uname -m
[  +42 ms] Exit code 0 from: uname -m
[   +1 ms] aarch64
[  +10 ms] executing: [Removed_Directory_Name_Sorry/flutter/] git -c log.showSignature=false log -n 1
--pretty=format:%H
[ +102 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] 484e09ef798bb6feaad5942b81bcc92baa061e52
[   +1 ms] executing: [Removed_Directory_Name_Sorry/flutter/] git tag --points-at
484e09ef798bb6feaad5942b81bcc92baa061e52
[  +84 ms] Exit code 0 from: git tag --points-at 484e09ef798bb6feaad5942b81bcc92baa061e52
[   +7 ms] executing: [Removed_Directory_Name_Sorry/flutter/] git describe --match *.*.* --long --tags
484e09ef798bb6feaad5942b81bcc92baa061e52
[  +95 ms] Exit code 0 from: git describe --match *.*.* --long --tags
484e09ef798bb6feaad5942b81bcc92baa061e52
[   +2 ms] 3.6.0-2.0.pre-55-g484e09ef79
[  +60 ms] executing: [Removed_Directory_Name_Sorry/flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[  +45 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] origin/master
[        ] executing: [Removed_Directory_Name_Sorry/flutter/] git ls-remote --get-url origin
[  +34 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +96 ms] executing: [Removed_Directory_Name_Sorry/flutter/] git rev-parse --abbrev-ref HEAD
[  +44 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms] master
[ +195 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[  +14 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +615 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[  +37 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[  +16 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +498 ms] Skipping pub get: version match.
[ +927 ms] Launching lib/main.dart on Web Server in debug mode...
[ +495 ms] Initializing file store
[  +49 ms] Skipping target: gen_localizations
[  +25 ms] Skipping target: gen_dart_plugin_registrant
[   +2 ms] Skipping target: _composite
[  +12 ms] complete
[   +9 ms] Updating assets
[ +480 ms] Waiting for connection from debug service on Web Server...
[ +640 ms] Compilation to SkSL failed.

Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag: GLSL to SPIRV
                    failed; Compilation error. 3 error(s) and 1 warning(s).

Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag: warning:
                    (version, profile) forced to be (460, core), while in source code it is (320,
es)

Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:9: error:
                    '#include' : Included file not found. for header name:
flutter/runtime_effect.glsl

Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:93: error:
                    'FlutterFragCoord' : no matching overloaded function found

Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:93: error: '='
                    :  cannot convert from ' const float' to ' temp 2-component vector of float'
                    
                    
                    
[ +100 ms] "flutter run" took 3,967ms.

[  +30 ms] Sending crash report to Google.
[+1950 ms] Crash report sent (report ID: 2cdb150de4d35983)
[  +11 ms] Oops; flutter has exited unexpectedly: "ShaderCompilerException: Shader compilation of
           "Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag" to
           "build/flutter_assets/shaders/ink_sparkle.frag" failed with exit code 1.
           impellerc stdout:
           
           impellerc stderr:
           Compilation to SkSL failed.
           Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:
GLSL to SPIRV
           failed; Compilation error. 3 error(s) and 1 warning(s).
           Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:
warning:
           (version, profile) forced to be (460, core), while in source code it is (320, es)
           Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:9:
error:
           '#include' : Included file not found. for header name: flutter/runtime_effect.glsl
           Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:93:
error:
           'FlutterFragCoord' : no matching overloaded function found
           Removed_Directory_Name_Sorry/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:93:
error: '='
           :  cannot convert from ' const float' to ' temp 2-component vector of float'
           
           
           
           
           ".
[ +155 ms] executing: [Removed_Directory_Name_Sorry/flutter/] git -c log.showSignature=false log -n 1
--pretty=format:%ar
[  +36 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%ar
[   +1 ms] 87 minutes ago
[+1968 ms] A crash report has been written to Removed_Directory_Name_Sorry/flutter_03.log.
[   +1 ms] This crash may already be reported. Check GitHub for similar crashes.
[   +2 ms]
https://github.com/flutter/flutter/issues?q=is%3Aissue+ShaderCompilerException%3A+Shader+compilatio
n+of+%22%2Fhome%2Fcsg%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fin
k_sparkle.frag%22+to+%22build%2Fflutter_assets%2Fshaders%2Fink_sparkle.frag%22+failed+with+exit+cod
e+1.%0Aimpellerc+stdout%3A%0A%0Aimpellerc+stderr%3A%0ACompilation+to+SkSL+failed.%0A%2Fhome%2Fcsg%2
Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sparkle.frag%3A+GLSL+
to+SPIRV+failed%3B+Compilation+error.+3+error%28s%29+and+1+warning%28s%29.%0A%2Fhome%2Fcsg%2Fcoding
%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sparkle.frag%3A+warning%3A+%
28version%2C+profile%29+forced+to+be+%28460%2C+core%29%2C+while+in+source+code+it+is+%28320%2C+es%2
9%0A%2Fhome%2Fcsg%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sp
arkle.frag%3A9%3A+error%3A+%27%23include%27+%3A+Included+file+not+found.+for+header+name%3A+flutter
%2Fruntime_effect.glsl%0A%2Fhome%2Fcsg%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmateri
al%2Fshaders%2Fink_sparkle.frag%3A93%3A+error%3A+%27FlutterFragCoord%27+%3A+no+matching+overloaded+
function+found%0A%2Fhome%2Fcsg%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fsha
ders%2Fink_sparkle.frag%3A93%3A+error%3A+%27%3D%27+%3A++cannot+convert+from+%27+const+float%27+to+%
27+temp+2-component+vector+of+float%27%0A%0A%0A%0A%0A
           
[   +9 ms] To report your crash to the Flutter team, first read the guide to filing a bug.
[        ] https://flutter.dev/docs/resources/bug-reports
           
[   +1 ms] Create a new GitHub issue by pasting this link into your browser and completing the
issue template.
           Thank you!
[ +111 ms]
https://github.com/flutter/flutter/issues/new?title=%5Btool_crash%5D+ShaderCompilerException&body=%
23%23+Command%0A%60%60%60%0Aflutter+run+--web-renderer+canvaskit+-d+web-server+-v%0A%60%60%60%0A%0A
%23%23+Steps+to+Reproduce%0A1.+...%0A2.+...%0A3.+...%0A%0A%23%23+Logs%0AShaderCompilerException%0A%
60%60%60%0A%230++++++ShaderCompiler.compileShader+%28package%3Aflutter_tools%2Fsrc%2Fbuild_system%2
Ftargets%2Fshader_compiler.dart%3A201%3A9%29%0A%3Casynchronous+suspension%3E%0A%231++++++writeBundl
e.%3Canonymous+closure%3E+%28package%3Aflutter_tools%2Fsrc%2Fbundle_builder.dart%3A185%3A25%29%0A%3
Casynchronous+suspension%3E%0A%232++++++Future.wait.%3Canonymous+closure%3E+%28dart%3Aasync%2Ffutur
e.dart%3A522%3A21%29%0A%3Casynchronous+suspension%3E%0A%233++++++writeBundle+%28package%3Aflutter_t
ools%2Fsrc%2Fbundle_builder.dart%3A163%3A3%29%0A%3Casynchronous+suspension%3E%0A%234++++++WebDevFS.
update+%28package%3Aflutter_tools%2Fsrc%2Fisolated%2Fdevfs_web.dart%3A867%3A9%29%0A%3Casynchronous+
suspension%3E%0A%235++++++ResidentWebRunner._updateDevFS+%28package%3Aflutter_tools%2Fsrc%2Fisolate
d%2Fresident_web_runner.dart%3A490%3A35%29%0A%3Casynchronous+suspension%3E%0A%236++++++ResidentWebR
unner.run.%3Canonymous+closure%3E+%28package%3Aflutter_tools%2Fsrc%2Fisolated%2Fresident_web_runner
.dart%3A284%3A41%29%0A%3Casynchronous+suspension%3E%0A%237++++++asyncGuard.%3Canonymous+closure%3E+
%28package%3Aflutter_tools%2Fsrc%2Fbase%2Fasync_guard.dart%3A111%3A24%29%0A%3Casynchronous+suspensi
on%3E%0A%60%60%60%0A%60%60%60%0A%1B%5B32m%5B%E2%9C%93%5D%1B%5B39m+Flutter+%28Channel+master%2C+3.6.
0-3.0.pre.55%2C+on+Debian+GNU%2FLinux+11+%28bullseye%29%0A++++5.10.136-19394-g7a24dee39fa0%2C+local
e+en_US.UTF-8%29%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Flutter+version+3.6.0-3.0.pre.55+on+channel+mast
er+at+%2Fhome%2Fcsg%2Fcoding%2Fflutter%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Upstream+repository+https%
3A%2F%2Fgithub.com%2Fflutter%2Fflutter.git%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Framework+revision+484
e09ef79+%2887+minutes+ago%29%2C+2022-11-16+05%3A18%3A10+-0500%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Eng
ine+revision+0241f18cbe%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Dart+version+2.19.0+%28build+2.19.0-406.0
.dev%29%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+DevTools+version+2.19.0%0A%0A%1B%5B32m%5B%E2%9C%93%5D%1B%
5B39m+Chrome+-+develop+for+the+web%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+CHROME_EXECUTABLE+%3D+%2Fusr%2
Fbin%2Fgarcon-url-handler%0A%0A%1B%5B33m%5B%21%5D%1B%5B39m+Flutter+IDE+Support+%28No+supported+IDEs
+installed%29%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+IntelliJ+-+https%3A%2F%2Fwww.jetbrains.com%2Fidea%2
F%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Android+Studio+-+https%3A%2F%2Fdeveloper.android.com%2Fstudio%2
F%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+VS+Code+-+https%3A%2F%2Fcode.visualstudio.com%2F%0A%0A%1B%5B32m
%5B%E2%9C%93%5D%1B%5B39m+Connected+device+%282+available%29%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Web+S
erver+%28web%29+%E2%80%A2+web-server+%E2%80%A2+web-javascript+%E2%80%A2+Flutter+Tools%0A++++%1B%5B3
2m%E2%80%A2%1B%5B39m+Chrome+%28web%29+++++%E2%80%A2+chrome+++++%E2%80%A2+web-javascript+%E2%80%A2+u
nknown%0A%0A%1B%5B32m%5B%E2%9C%93%5D%1B%5B39m+HTTP+Host+Availability%0A++++%1B%5B32m%E2%80%A2%1B%5B
39m+All+required+HTTP+hosts+are+available%0A%0A%1B%5B33m%21%1B%5B39m+Doctor+found+issues+in+1+categ
ory.%0A%0A%60%60%60%0A%0A%23%23+Flutter+Application+Metadata%0A%2A%2AType%2A%2A%3A+app%0A%2A%2AVers
ion%2A%2A%3A+1.0.0%2B1%0A%2A%2AMaterial%2A%2A%3A+true%0A%2A%2AAndroid+X%2A%2A%3A+false%0A%2A%2AModu
le%2A%2A%3A+false%0A%2A%2APlugin%2A%2A%3A+false%0A%2A%2AAndroid+package%2A%2A%3A+null%0A%2A%2AiOS+b
undle+identifier%2A%2A%3A+null%0A%2A%2ACreation+channel%2A%2A%3A+master%0A%2A%2ACreation+framework+
version%2A%2A%3A+92df0264036df2b2d4a243aac1b147c1ebdab0fb%0A%0A&labels=tool%2Csevere%3A+crash
                    
[  +16 ms] Running 1 shutdown hook
[   +3 ms] Shutdown hooks complete
[   +1 ms] exiting with code 1
```

---

## Flutter Application Metadata
**Type**: app
**Version**: 1.0.0+1
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: master
**Creation framework version**: 92df0264036df2b2d4a243aac1b147c1ebdab0fb

---

## Further Edits

> I already upgraded to `Flutter (Channel master, 3.6.0-5.0.pre.31, on Debian GNU/Linux 11
    (bullseye) 5.10.136-19394-g7a24dee39fa0, locale en_US.UTF-8)`...
It still throws the same error, looks like I can't use this for a while longer.

> This is ridiculous.
I already upgraded (again) to `Flutter (Channel master, 3.6.0-6.0.pre.28, on Debian GNU/Linux 11 (bullseye)
    5.10.136-19394-g7a24dee39fa0, locale en_US.UTF-8)`, same error smacks me in the face.

**Comment by huycozy** [comment]
Hi @JumperBot 
I see that there is an issue that has been fixed recently #107590, but I also can see you are running the project on master `3.6.0-3.0.pre.55`. From my side, I can not reproduce this issue on `3.6.0-4.0.pre.28`, `Ubuntu 22.04.1 LTS 5.15.0-52-generic`. 

Does this issue happen on the Flutter default sample project (counter app)? And can you try executing `flutter config --enable-web` and then run it again?

**Comment by JumperBot** [comment]
Hello there @huycozy
Nope, it doesn't work either; still the same `SkSL compilation error`.
Also, `web` is the only platform enabled in my `Flutter` config.

**Comment by JumperBot** [comment]
I am currently upgrading to `3.6.0-3.0.pre`, hope it works!

**Comment by JumperBot** [comment]
I already upgraded, yet the issue still persists even on the default `counter app`.

**Comment by huycozy** [comment]
@JumperBot Thanks for providing more info. Labeling this issue for further investigation from the team.

/cc @christopherfujino

**Comment by JumperBot** [comment]
I already upgraded to `Flutter (Channel master, 3.6.0-5.0.pre.31, on Debian GNU/Linux 11
    (bullseye) 5.10.136-19394-g7a24dee39fa0, locale en_US.UTF-8)`...
It still throws the same error, looks like I can't use this for a while longer.

**Comment by JumperBot** [comment]
This is ridiculous.
I already upgraded (again) to `Flutter (Channel master, 3.6.0-6.0.pre.28, on Debian GNU/Linux 11 (bullseye)
    5.10.136-19394-g7a24dee39fa0, locale en_US.UTF-8)`, same error smacks me in the face.

**Comment by jonahwilliams** [comment]
@JumperBot just to confirm, are you running on ChromeOS?

**Comment by JumperBot** [comment]
> @JumperBot just to confirm, are you running on ChromeOS?

Yes I am on the Crostini container, is this an isolated issue?

**Comment by jimmy-vo** [comment]
I can reproduce it on my the new Macbook pro M1 (just arrived) in an ubuntu:20.04 based container. It works in the host environment (Macbook) without any problem. The build also works fine last night on another local machine (windows 11), and a ubuntu:20.04 based container (also works with github action environment). The only different I can tell is I had to remove `lib32stdc++6` from `apt-get install` in dockerfile on the mac because it cannot find the package, not sure why yet, but I will look into it later

**Comment by TahaTesser** [comment]
Running into this issue on macOS when running the app on an Android emulator on the latest beta
<details> 
<summary>logs</summary> 

```dart
tahatesser@Tahas-MacBook-Pro plants % fr
Running "flutter pub get" in plants...                             230ms
Resolving dependencies...
  string_scanner 1.1.1 (1.2.0 available)
  test_api 0.4.15 (0.4.16 available)
Got dependencies!
Using hardware rendering with device sdk gphone64 arm64. If you notice graphics
artifacts, consider enabling software rendering with
"--enable-software-rendering".
Launching lib/main.dart on sdk gphone64 arm64 in debug mode...

Target debug_android_application failed: ShaderCompilerException: Shader compilation of "/Users/tahatesser/Code/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag" to "/Users/tahatesser/StudioProjects/plants/build/app/intermediates/flutter/debug/flutter_assets/shaders/ink_sparkle.frag" failed with exit code -9.
impellerc stdout:

impellerc stderr:





FAILURE: Build failed with an exception.

* Where:
Script '/Users/tahatesser/Code/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 1151

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command '/Users/tahatesser/Code/flutter/bin/flutter'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 30s
Running Gradle task 'assembleDebug'...                             31.5s
Exception: Gradle task assembleDebug failed with exit code 1
``` 
	
</details>

<details> 
<summary>flutter doctor -v</summary> 

```dart
[!] Flutter (Channel beta, 3.6.0-0.1.pre, on macOS 13.0.1 22A400 darwin-arm64, locale en-EE)
    • Flutter version 3.6.0-0.1.pre on channel beta at /Users/tahatesser/Code/flutter
    ! Upstream repository https://github.com/flutter/flutter.git is not the same as FLUTTER_GIT_URL
    • FLUTTER_GIT_URL = git@github.com:NevercodeHQ/flutter.git
    • Framework revision 75927305ff (7 days ago), 2022-11-17 05:52:50 +0530
    • Engine revision b754eabc98
    • Dart version 2.19.0 (build 2.19.0-374.1.beta)
    • DevTools version 2.19.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to
      perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/tahatesser/Code/android-sdk
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/tahatesser/Code/android-sdk
    • Java binary at: /Users/tahatesser/Code/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment Temurin-16.0.2+7 (build 16.0.2+7)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.73.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554 • android-arm64  • Android 13 (API 33) (emulator)
    • macOS (desktop)             • macos         • darwin-arm64   • macOS 13.0.1 22A400 darwin-arm64
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
``` 
	
</details>

**Comment by cjsmocjsmo** [comment]
I am running into this problem on Ubuntu 22.04 ARM64 the logs and flutter doctor -v can be found at #115873

**Comment by JumperBot** [comment]
9 days ago, `Flutter` worked so flawlessly on my machine.
Now, I could not even work on my current projects.
I might be forced to either downgrade or abandon these altogether and start again.
I hope that this would be fixed by December as a gift from the devs :)

**Comment by JumperBot** [comment]
I have already rollbacked to f9972818f4e8bf464b378f0942a153fa391a0b7a using:
```
git checkout f9972818f4e8bf464b378f0942a153fa391a0b7a
flutter doctor
```
![Screenshot 2022-11-26 11 53 30](https://user-images.githubusercontent.com/88363421/204071641-4e26c70c-25d4-4a05-86f9-ffa72ac074f9.png)

---

And then I ran:
```
flutter run --web-renderer canvaskit -d web-server
```
![image](https://user-images.githubusercontent.com/88363421/204071661-0a500e5f-80fc-4f2a-a0d6-15cda8ad8b2a.png)

---

@jonahwilliams, I know you might be busy right now,
but maybe there was an overlooked change right after f9972818f4e8bf464b378f0942a153fa391a0b7a?
For now, I'll be using this rollbacked version for development before I with everyone else loses their sanity.

---

I will continue on checking other versions upstream to check where did this all went wrong.

**Comment by JumperBot** [comment]
Updates:
- 3a298d0b95e5d1c2063ef4fc2260bb3e1a93a788 brings the same issue.

**Comment by JumperBot** [comment]
Hello @jonahwilliams 

'Just want to help with debugging.
Flutter still works on 9d64a0f40bc0d9d9fbc88f290b4f7e52fd5013cb but it breaks at e9c6212da02ab7edc39cc0295f72df4a40f0d2ba.

And as the errors back-up,
![Screenshot 2022-11-26 19 45 39](https://user-images.githubusercontent.com/88363421/204087160-bdfd42d1-35b4-4ea1-82c0-ff8b71641758.png)

It looks like this is an issue directly produced by #114214.

---

That's just my findings,
I just really want to upgrade Flutter on my side now to contrinue on my normal life.

**Comment by jonahwilliams** [comment]
I landed a small change that adds more logging to the failing command, if you're on master and experiencing this issue, please update past https://github.com/flutter/flutter/pull/116123 and upload logs of a verbose run where this fails. If you could also include your host platform (windows, macOS, etc) and any special configurations you might be using (WSL) that would be helpful.

**Comment by JumperBot** [comment]
Hi @jonahwilliams , here's the much more verbose "verbose" log from this command:
❯ flutter run --web-renderer canvaskit -d web-server --verbose

```
[ +102 ms] executing: uname -m
[  +47 ms] Exit code 0 from: uname -m
[        ] aarch64
[   +4 ms] executing: [/home/coding/flutter/] git -c log.showSignature=false log -n 1
--pretty=format:%H
[  +36 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 31ea315c3c4142265a6e1efe065d751e15e6eaf2
[        ] executing: [/home/coding/flutter/] git tag --points-at
31ea315c3c4142265a6e1efe065d751e15e6eaf2
[  +61 ms] Exit code 0 from: git tag --points-at 31ea315c3c4142265a6e1efe065d751e15e6eaf2
[   +7 ms] executing: [/home/coding/flutter/] git describe --match *.*.* --long --tags
31ea315c3c4142265a6e1efe065d751e15e6eaf2
[ +120 ms] Exit code 0 from: git describe --match *.*.* --long --tags
31ea315c3c4142265a6e1efe065d751e15e6eaf2
[  +11 ms] 3.6.0-10.0.pre-25-g31ea315c3c
[ +110 ms] executing: [/home/coding/flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[  +32 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[   +1 ms] origin/master
[        ] executing: [/home/coding/flutter/] git ls-remote --get-url origin
[  +28 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] https://github.com/flutter/flutter.git
[  +72 ms] executing: [/home/coding/flutter/] git rev-parse --abbrev-ref HEAD
[  +24 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[ +160 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[  +11 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +347 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +9 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +529 ms] Skipping pub get: version match.
[ +863 ms] Launching lib/main.dart on Web Server in debug mode...
[ +613 ms] Initializing file store
[  +60 ms] Skipping target: gen_localizations
[  +21 ms] Skipping target: gen_dart_plugin_registrant
[   +2 ms] Skipping target: _composite
[   +9 ms] complete
[   +6 ms] Updating assets
[ +341 ms] Waiting for connection from debug service on Web Server...
[ +154 ms] shaderc command:
[/home/coding/flutter/bin/cache/artifacts/engine/linux-arm64/impellerc, --sksl, --iplr, --json,
--sl=build/flutter_assets/shaders/ink_sparkle.frag,
--spirv=build/flutter_assets/shaders/ink_sparkle.frag.spirv,
--input=/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag,
--input-type=frag, --include=/home/coding/flutter/packages/flutter/lib/src/material/shaders,
--include=/home/coding/flutter/bin/cache/artifacts/engine/linux-arm64/shader_lib]
[ +350 ms] Compilation to SkSL failed.

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag: GLSL to SPIRV
                    failed; Compilation error. 3 error(s) and 1 warning(s).

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag: warning:
                    (version, profile) forced to be (460, core), while in source code it is (320,
es)

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:9: error:
                    '#include' : Included file not found. for header name:
flutter/runtime_effect.glsl

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:93: error:
                    'FlutterFragCoord' : no matching overloaded function found

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:93: error: '='
                    :  cannot convert from ' const float' to ' temp 2-component vector of float'
                    
                    
                    
[  +83 ms] "flutter run" took 3,445ms.

[  +27 ms] Sending crash report to Google.
[+1767 ms] Crash report sent (report ID: c1dcf211891d3a81)
[   +9 ms] Oops; flutter has exited unexpectedly: "ShaderCompilerException: Shader compilation of
           "/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag" to
           "build/flutter_assets/shaders/ink_sparkle.frag" failed with exit code 1.
           impellerc stdout:
           
           impellerc stderr:
           Compilation to SkSL failed.
           /home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:
GLSL to SPIRV
           failed; Compilation error. 3 error(s) and 1 warning(s).
           /home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:
warning:
           (version, profile) forced to be (460, core), while in source code it is (320, es)
           /home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:9:
error:
           '#include' : Included file not found. for header name: flutter/runtime_effect.glsl
           /home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:93:
error:
           'FlutterFragCoord' : no matching overloaded function found
           /home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag:93:
error: '='
           :  cannot convert from ' const float' to ' temp 2-component vector of float'
           
           
           
           
           ".
[ +116 ms] executing: [/home/coding/flutter/] git -c log.showSignature=false log -n 1
--pretty=format:%ar
[  +28 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%ar
[        ] 17 minutes ago
[+3284 ms] A crash report has been written to /home/coding/REDACTED/flutter_09.log.
[   +1 ms] This crash may already be reported. Check GitHub for similar crashes.
[  +31 ms]
https://github.com/flutter/flutter/issues?q=is%3Aissue+ShaderCompilerException%3A+Shader+compilatio
n+of+%22%2Fhome%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fin
k_sparkle.frag%22+to+%22build%2Fflutter_assets%2Fshaders%2Fink_sparkle.frag%22+failed+with+exit+cod
e+1.%0Aimpellerc+stdout%3A%0A%0Aimpellerc+stderr%3A%0ACompilation+to+SkSL+failed.%0A%2Fhome%2
Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sparkle.frag%3A+GLSL+
to+SPIRV+failed%3B+Compilation+error.+3+error%28s%29+and+1+warning%28s%29.%0A%2Fhome%2Fcoding
%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sparkle.frag%3A+warning%3A+%
28version%2C+profile%29+forced+to+be+%28460%2C+core%29%2C+while+in+source+code+it+is+%28320%2C+es%2
9%0A%2Fhome%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sp
arkle.frag%3A9%3A+error%3A+%27%23include%27+%3A+Included+file+not+found.+for+header+name%3A+flutter
%2Fruntime_effect.glsl%0A%2Fhome%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmateri
al%2Fshaders%2Fink_sparkle.frag%3A93%3A+error%3A+%27FlutterFragCoord%27+%3A+no+matching+overloaded+
function+found%0A%2Fhome%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fsha
ders%2Fink_sparkle.frag%3A93%3A+error%3A+%27%3D%27+%3A++cannot+convert+from+%27+const+float%27+to+%
27+temp+2-component+vector+of+float%27%0A%0A%0A%0A%0A
           
[  +26 ms] To report your crash to the Flutter team, first read the guide to filing a bug.
[   +1 ms] https://flutter.dev/docs/resources/bug-reports
           
[   +2 ms] Create a new GitHub issue by pasting this link into your browser and completing the
issue template.
           Thank you!
[  +95 ms]
https://github.com/flutter/flutter/issues/new?title=%5Btool_crash%5D+ShaderCompilerException&body=%
23%23+Command%0A%60%60%60%0Aflutter+run+--web-renderer+canvaskit+-d+web-server+--verbose%0A%60%60%6
0%0A%0A%23%23+Steps+to+Reproduce%0A1.+...%0A2.+...%0A3.+...%0A%0A%23%23+Logs%0AShaderCompilerExcept
ion%0A%60%60%60%0A%230++++++ShaderCompiler.compileShader+%28package%3Aflutter_tools%2Fsrc%2Fbuild_s
ystem%2Ftargets%2Fshader_compiler.dart%3A201%3A9%29%0A%3Casynchronous+suspension%3E%0A%231++++++wri
teBundle.%3Canonymous+closure%3E+%28package%3Aflutter_tools%2Fsrc%2Fbundle_builder.dart%3A185%3A25%
29%0A%3Casynchronous+suspension%3E%0A%232++++++Future.wait.%3Canonymous+closure%3E+%28dart%3Aasync%
2Ffuture.dart%3A522%3A21%29%0A%3Casynchronous+suspension%3E%0A%233++++++writeBundle+%28package%3Afl
utter_tools%2Fsrc%2Fbundle_builder.dart%3A163%3A3%29%0A%3Casynchronous+suspension%3E%0A%234++++++We
bDevFS.update+%28package%3Aflutter_tools%2Fsrc%2Fisolated%2Fdevfs_web.dart%3A867%3A9%29%0A%3Casynch
ronous+suspension%3E%0A%235++++++ResidentWebRunner._updateDevFS+%28package%3Aflutter_tools%2Fsrc%2F
isolated%2Fresident_web_runner.dart%3A487%3A35%29%0A%3Casynchronous+suspension%3E%0A%236++++++Resid
entWebRunner.run.%3Canonymous+closure%3E+%28package%3Aflutter_tools%2Fsrc%2Fisolated%2Fresident_web
_runner.dart%3A284%3A41%29%0A%3Casynchronous+suspension%3E%0A%237++++++asyncGuard.%3Canonymous+clos
ure%3E+%28package%3Aflutter_tools%2Fsrc%2Fbase%2Fasync_guard.dart%3A111%3A24%29%0A%3Casynchronous+s
uspension%3E%0A%60%60%60%0A%60%60%60%0A%1B%5B32m%5B%E2%9C%93%5D%1B%5B39m+Flutter+%28Channel+master%
2C+3.6.0-11.0.pre.25%2C+on+Debian+GNU%2FLinux+11+%28bullseye%29%0A++++5.10.136-19394-g7a24dee39fa0%
2C+locale+en_US.UTF-8%29%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Flutter+version+3.6.0-11.0.pre.25+on+cha
nnel+master+at+%2Fhome%2Fcoding%2Fflutter%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Upstream+reposito
ry+https%3A%2F%2Fgithub.com%2Fflutter%2Fflutter.git%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Framework+rev
ision+31ea315c3c+%2817+minutes+ago%29%2C+2022-11-29+18%3A29%3A02+-0800%0A++++%1B%5B32m%E2%80%A2%1B%
5B39m+Engine+revision+25a2ac85f6%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Dart+version+2.19.0+%28build+2.1
9.0-430.0.dev%29%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+DevTools+version+2.19.0%0A%0A%1B%5B32m%5B%E2%9C%
93%5D%1B%5B39m+Chrome+-+develop+for+the+web%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+CHROME_EXECUTABLE+%3D
+%2Fusr%2Fbin%2Fgarcon-url-handler%0A%0A%1B%5B33m%5B%21%5D%1B%5B39m+Flutter+IDE+Support+%28No+suppo
rted+IDEs+installed%29%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+IntelliJ+-+https%3A%2F%2Fwww.jetbrains.com
%2Fidea%2F%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Android+Studio+-+https%3A%2F%2Fdeveloper.android.com%2
Fstudio%2F%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+VS+Code+-+https%3A%2F%2Fcode.visualstudio.com%2F%0A%0A
%1B%5B32m%5B%E2%9C%93%5D%1B%5B39m+Connected+device+%282+available%29%0A++++%1B%5B32m%E2%80%A2%1B%5B
39m+Web+Server+%28web%29+%E2%80%A2+web-server+%E2%80%A2+web-javascript+%E2%80%A2+Flutter+Tools%0A++
++%1B%5B32m%E2%80%A2%1B%5B39m+Chrome+%28web%29+++++%E2%80%A2+chrome+++++%E2%80%A2+web-javascript+%E
2%80%A2+unknown%0A%0A%1B%5B32m%5B%E2%9C%93%5D%1B%5B39m+HTTP+Host+Availability%0A++++%1B%5B32m%E2%80
%A2%1B%5B39m+All+required+HTTP+hosts+are+available%0A%0A%1B%5B33m%21%1B%5B39m+Doctor+found+issues+i
n+1+category.%0A%0A%60%60%60%0A%0A%23%23+Flutter+Application+Metadata%0A%2A%2AType%2A%2A%3A+app%0A%
2A%2AVersion%2A%2A%3A+1.0.0%2B1%0A%2A%2AMaterial%2A%2A%3A+true%0A%2A%2AAndroid+X%2A%2A%3A+false%0A%
2A%2AModule%2A%2A%3A+false%0A%2A%2APlugin%2A%2A%3A+false%0A%2A%2AAndroid+package%2A%2A%3A+null%0A%2
A%2AiOS+bundle+identifier%2A%2A%3A+null%0A%2A%2ACreation+channel%2A%2A%3A+master%0A%2A%2ACreation+f
ramework+version%2A%2A%3A+92df0264036df2b2d4a243aac1b147c1ebdab0fb%0A%0A&labels=tool%2Csevere%3A+cr
ash
           
[  +20 ms] Running 1 shutdown hook
[   +3 ms] Shutdown hooks complete
[   +9 ms] exiting with code 1
```

**Comment by JumperBot** [comment]
Hello again, @jonahwilliams 

I ran it again after upgrading and a quick `flutter clean`:

```
[  +95 ms] executing: uname -m
[  +45 ms] Exit code 0 from: uname -m
[   +1 ms] aarch64
[   +5 ms] executing: [/home/coding/flutter/] git -c
log.showSignature=false log -n 1 --pretty=format:%H
[  +59 ms] Exit code 0 from: git -c log.showSignature=false log -n 1
--pretty=format:%H
[   +1 ms] b75f1a941b6d007c4bdcde8b3acc9670d2a85292
[        ] executing: [/home/coding/flutter/] git tag --points-at
b75f1a941b6d007c4bdcde8b3acc9670d2a85292
[  +46 ms] Exit code 0 from: git tag --points-at
b75f1a941b6d007c4bdcde8b3acc9670d2a85292
[  +14 ms] executing: [/home/coding/flutter/] git describe --match
*.*.* --long --tags b75f1a941b6d007c4bdcde8b3acc9670d2a85292
[  +77 ms] Exit code 0 from: git describe --match *.*.* --long --tags
b75f1a941b6d007c4bdcde8b3acc9670d2a85292
[   +3 ms] 3.7.0-1.0.pre-24-gb75f1a941b
[  +66 ms] executing: [/home/coding/flutter/] git rev-parse
--abbrev-ref --symbolic @{upstream}
[  +16 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic
@{upstream}
[   +2 ms] origin/master
[        ] executing: [/home/coding/flutter/] git ls-remote --get-url
origin
[  +21 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] https://github.com/flutter/flutter.git
[ +101 ms] executing: [/home/coding/flutter/] git rev-parse
--abbrev-ref HEAD
[  +30 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] master
[ +176 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'IOSEngineArtifacts' is not required,
skipping update.
[   +1 ms] Artifact Instance of 'FlutterWebSdk' is not required, skipping
update.
[  +13 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required,
skipping update.
[  +10 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required,
skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required,
skipping update.
[   +1 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not
required, skipping update.
[   +2 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not
required, skipping update.
[ +287 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'IOSEngineArtifacts' is not required,
skipping update.
[   +9 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required,
skipping update.
[   +1 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required,
skipping update.
[   +1 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required,
skipping update.
[   +5 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not
required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not
required, skipping update.
[ +411 ms] Skipping pub get: version match.
[ +633 ms] Launching lib/main.dart on Web Server in debug mode...
[ +428 ms] Initializing file store
[  +49 ms] Skipping target: gen_localizations
[  +32 ms] Skipping target: gen_dart_plugin_registrant
[   +2 ms] Skipping target: _composite
[  +15 ms] complete
[   +6 ms] Updating assets
[ +360 ms] Waiting for connection from debug service on Web Server...
[ +155 ms] shaderc command:
[/home/coding/flutter/bin/cache/artifacts/engine/linux-arm64/impellerc
, --sksl, --iplr, --json,
--sl=build/flutter_assets/shaders/ink_sparkle.frag,
--spirv=build/flutter_assets/shaders/ink_sarkle.frag.spirv,
--input=/home/coding/flutter/packages/flutter/lib/src/material/shaders
/ink_sparkle.frag, --input-type=frag,
--include=/home/coding/flutter/packages/flutter/lib/src/material/shade
rs,
--include=/home/coding/flutter/bin/cache/artifacts/engine/linux-arm64/
shader_lib]
[ +377 ms] Compilation to SkSL failed.

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
                    rkle.frag: GLSL to SPIRV failed; Compilation error. 3
error(s) and 1
                    warning(s).

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
                    rkle.frag: warning: (version, profile) forced to be
(460, core), while in
                    source code it is (320, es)

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
                    rkle.frag:9: error: '#include' : Included file not
found. for header name:
                    flutter/runtime_effect.glsl

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
                    rkle.frag:93: error: 'FlutterFragCoord' : no matching
overloaded function
                    found

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
                    rkle.frag:93: error: '=' :  cannot convert from '
const float' to ' temp
                    2-component vector of float'
                    
                    
                    
[  +90 ms] "flutter run" took 2,928ms.

[  +33 ms] Sending crash report to Google.
[+1779 ms] Crash report sent (report ID: 0cfc243ff2d2d3c0)
[  +10 ms] Oops; flutter has exited unexpectedly:
"ShaderCompilerException: Shader
           compilation of

"/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_sp
           arkle.frag" to "build/flutter_assets/shaders/ink_sparkle.frag"
failed with
           exit code 1.
           impellerc stdout:
           
           impellerc stderr:
           Compilation to SkSL failed.

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
           rkle.frag: GLSL to SPIRV failed; Compilation error. 3 error(s)
and 1
           warning(s).

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
           rkle.frag: warning: (version, profile) forced to be (460,
core), while in
           source code it is (320, es)

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
           rkle.frag:9: error: '#include' : Included file not found. for
header name:
           flutter/runtime_effect.glsl

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
           rkle.frag:93: error: 'FlutterFragCoord' : no matching
overloaded function
           found

/home/coding/flutter/packages/flutter/lib/src/material/shaders/ink_spa
           rkle.frag:93: error: '=' :  cannot convert from ' const float'
to ' temp
           2-component vector of float'
           
           
           
           
           ".
[ +103 ms] executing: [/home/coding/flutter/] git -c
log.showSignature=false log -n 1 --pretty=format:%ar
[  +20 ms] Exit code 0 from: git -c log.showSignature=false log -n 1
--pretty=format:%ar
[   +1 ms] 73 minutes ago
[+1550 ms] A crash report has been written to
                    /home/coding/REDACTED/new/flutter_02.log.
[        ] This crash may already be reported. Check GitHub for similar
crashes.
[   +2 ms]
https://github.com/flutter/flutter/issues?q=is%3Aissue+ShaderCompilerExcep
tion%3A+Shader+compilation+of+%22%2Fhome%2Fcoding%2Fflutter%2Fpackag
es%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sparkle.frag%22+to+%22
build%2Fflutter_assets%2Fshaders%2Fink_sparkle.frag%22+failed+with+exit+co
de+1.%0Aimpellerc+stdout%3A%0A%0Aimpellerc+stderr%3A%0ACompilation+to+SkSL
+failed.%0A%2Fhome%2Fcoding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fs
rc%2Fmaterial%2Fshaders%2Fink_sparkle.frag%3A+GLSL+to+SPIRV+failed%3B+Comp
ilation+error.+3+error%28s%29+and+1+warning%28s%29.%0A%2Fhome%2Fcodi
ng%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_s
parkle.frag%3A+warning%3A+%28version%2C+profile%29+forced+to+be+%28460%2C+
core%29%2C+while+in+source+code+it+is+%28320%2C+es%29%0A%2Fhome%2Fco
ding%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink
_sparkle.frag%3A9%3A+error%3A+%27%23include%27+%3A+Included+file+not+found
.+for+header+name%3A+flutter%2Fruntime_effect.glsl%0A%2Fhome%2Fcodin
g%2Fflutter%2Fpackages%2Fflutter%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sp
arkle.frag%3A93%3A+error%3A+%27FlutterFragCoord%27+%3A+no+matching+overloa
ded+function+found%0A%2Fhome%2Fcoding%2Fflutter%2Fpackages%2Fflutter
%2Flib%2Fsrc%2Fmaterial%2Fshaders%2Fink_sparkle.frag%3A93%3A+error%3A+%27%
3D%27+%3A++cannot+convert+from+%27+const+float%27+to+%27+temp+2-component+
vector+of+float%27%0A%0A%0A%0A%0A
           
[   +8 ms] To report your crash to the Flutter team, first read the guide
to filing a
           bug.
[   +1 ms] https://flutter.dev/docs/resources/bug-reports
           
[   +1 ms] Create a new GitHub issue by pasting this link into your
browser and
           completing the issue template. Thank you!
[  +62 ms]
https://github.com/flutter/flutter/issues/new?title=%5Btool_crash%5D+Shade
rCompilerException&body=%23%23+Command%0A%60%60%60%0Aflutter+run+--web-ren
derer+canvaskit+-d+web-server+--verbose%0A%60%60%60%0A%0A%23%23+Steps+to+R
eproduce%0A1.+...%0A2.+...%0A3.+...%0A%0A%23%23+Logs%0AShaderCompilerExcep
tion%0A%60%60%60%0A%230++++++ShaderCompiler.compileShader+%28package%3Aflu
tter_tools%2Fsrc%2Fbuild_system%2Ftargets%2Fshader_compiler.dart%3A201%3A9
%29%0A%3Casynchronous+suspension%3E%0A%231++++++writeBundle.%3Canonymous+c
losure%3E+%28package%3Aflutter_tools%2Fsrc%2Fbundle_builder.dart%3A185%3A2
5%29%0A%3Casynchronous+suspension%3E%0A%232++++++Future.wait.%3Canonymous+
closure%3E+%28dart%3Aasync%2Ffuture.dart%3A522%3A21%29%0A%3Casynchronous+s
uspension%3E%0A%233++++++writeBundle+%28package%3Aflutter_tools%2Fsrc%2Fbu
ndle_builder.dart%3A163%3A3%29%0A%3Casynchronous+suspension%3E%0A%234+++++
+WebDevFS.update+%28package%3Aflutter_tools%2Fsrc%2Fisolated%2Fdevfs_web.d
art%3A868%3A9%29%0A%3Casynchronous+suspension%3E%0A%235++++++ResidentWebRu
nner._updateDevFS+%28package%3Aflutter_tools%2Fsrc%2Fisolated%2Fresident_w
eb_runner.dart%3A487%3A35%29%0A%3Casynchronous+suspension%3E%0A%236++++++R
esidentWebRunner.run.%3Canonymous+closure%3E+%28package%3Aflutter_tools%2F
src%2Fisolated%2Fresident_web_runner.dart%3A284%3A41%29%0A%3Casynchronous+
suspension%3E%0A%237++++++asyncGuard.%3Canonymous+closure%3E+%28package%3A
flutter_tools%2Fsrc%2Fbase%2Fasync_guard.dart%3A111%3A24%29%0A%3Casynchron
ous+suspension%3E%0A%60%60%60%0A%60%60%60%0A%1B%5B32m%5B%E2%9C%93%5D%1B%5B
39m+Flutter+%28Channel+master%2C+3.7.0-2.0.pre.24%2C+on+Debian+GNU%2FLinux
+11%0A++++%28bullseye%29+5.10.136-19394-g7a24dee39fa0%2C+locale+en_US.UTF-
8%29%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Flutter+version+3.7.0-2.0.pre.24+on
+channel+master+at%0A++++++%2Fhome%2Fcoding%2Fflutter%0A++++%1B%5B32
m%E2%80%A2%1B%5B39m+Upstream+repository+https%3A%2F%2Fgithub.com%2Fflutter
%2Fflutter.git%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Framework+revision+b75f1a
941b+%2873+minutes+ago%29%2C+2022-12-02+18%3A32%3A24%0A++++++-0500%0A++++%
1B%5B32m%E2%80%A2%1B%5B39m+Engine+revision+afdc32933b%0A++++%1B%5B32m%E2%8
0%A2%1B%5B39m+Dart+version+2.19.0+%28build+2.19.0-444.0.dev%29%0A++++%1B%5
B32m%E2%80%A2%1B%5B39m+DevTools+version+2.20.0%0A%0A%1B%5B32m%5B%E2%9C%93%
5D%1B%5B39m+Chrome+-+develop+for+the+web%0A++++%1B%5B32m%E2%80%A2%1B%5B39m
+CHROME_EXECUTABLE+%3D+%2Fusr%2Fbin%2Fgarcon-url-handler%0A%0A%1B%5B33m%5B
%21%5D%1B%5B39m+Flutter+IDE+Support+%28No+supported+IDEs+installed%29%0A++
++%1B%5B32m%E2%80%A2%1B%5B39m+IntelliJ+-+https%3A%2F%2Fwww.jetbrains.com%2
Fidea%2F%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Android+Studio+-+https%3A%2F%2F
developer.android.com%2Fstudio%2F%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+VS+Cod
e+-+https%3A%2F%2Fcode.visualstudio.com%2F%0A%0A%1B%5B32m%5B%E2%9C%93%5D%1
B%5B39m+Connected+device+%282+available%29%0A++++%1B%5B32m%E2%80%A2%1B%5B3
9m+Web+Server+%28web%29+%E2%80%A2+web-server+%E2%80%A2+web-javascript+%E2%
80%A2+Flutter+Tools%0A++++%1B%5B32m%E2%80%A2%1B%5B39m+Chrome+%28web%29++++
+%E2%80%A2+chrome+++++%E2%80%A2+web-javascript+%E2%80%A2+unknown%0A%0A%1B%
5B32m%5B%E2%9C%93%5D%1B%5B39m+HTTP+Host+Availability%0A++++%1B%5B32m%E2%80
%A2%1B%5B39m+All+required+HTTP+hosts+are+available%0A%0A%1B%5B33m%21%1B%5B
39m+Doctor+found+issues+in+1+category.%0A%0A%60%60%60%0A%0A%23%23+Flutter+
Application+Metadata%0A%2A%2AType%2A%2A%3A+app%0A%2A%2AVersion%2A%2A%3A+1.
0.0%2B1%0A%2A%2AMaterial%2A%2A%3A+true%0A%2A%2AAndroid+X%2A%2A%3A+false%0A
%2A%2AModule%2A%2A%3A+false%0A%2A%2APlugin%2A%2A%3A+false%0A%2A%2AAndroid+
package%2A%2A%3A+null%0A%2A%2AiOS+bundle+identifier%2A%2A%3A+null%0A%2A%2A
Creation+channel%2A%2A%3A+master%0A%2A%2ACreation+framework+version%2A%2A%
3A+92df0264036df2b2d4a243aac1b147c1ebdab0fb%0A%0A&labels=tool%2Csevere%3A+
crash
           
[  +12 ms] Running 1 shutdown hook
[   +4 ms] Shutdown hooks complete
[   +1 ms] exiting with code 1
```

Hmm, it is what it is.
I am patient enough to wait until this is fixed though.

**Comment by jonahwilliams** [comment]
We're missing this artifact for arm64. Fix is https://flutter-review.git.corp.google.com/c/recipes/+/36921

We'll still disable this for web usage for now since we're blocked on CanvasKit changes

**Comment by jonahwilliams** [comment]
We believe this is fixed now, closing

**Comment by JumperBot** [comment]
@jonahwilliams Yup, that was a perfect fix, thanks!

**Comment by jimmy-vo** [comment]
the latest works on my M1 pro too. Thank you

**Comment by JumperBot** [comment]
After a bit of inactivity, it seems like the same error has happened again on 17b4c70ff5.
Maybe it's another issue connected with the Arm64 artifact?
